codalab/lib/bundle_cli.py

```diff
@@ -1372,6 +1372,26 @@ def do_upload_command(self, args):
             'metadata': metadata,
         }
 
+        # Validate store metadata.
+        store = metadata.get('store') or ''
```

Comment:
Thanks! I wonder -- in order to make this faster and simpler -- could we:
- not call `client.fetch('bundle_stores')` (which is an extra network request), but just catch the exception when we call the upload function?
- no need to print out available bundle stores on error, as a user can already do that with `cl store ls`. We could give a message "use cl store ls to view available bundle stores" if you'd like.

Reply:
@epicfaace from what I can see in `json_api_client.py`, the error that comes back is a generic 500 internal server error. So that that point, we don't know that it was specifically an incorrect store name that caused the error. Lmk if I'm looking in the wrong place.

If you're suggesting that we just catch the 500 and throw something generic (perhaps including `cl store ls` instructions), that seems fairly unhelpful for the user. However if you think that the extra `bundle_stores` call is going to be problematic performance-wise, I can make this change. Lmk!

Reply:
Perhaps we should change the backend code (https://github.com/codalab/codalab-worksheets/blob/master/codalab/rest/bundles.py#L1109) so it throws a 404 not found with a specific error about bundle store not found? Then we could catch it.

My main concern is that we're adding a new step where the user sees the "Validating bundle store..." message, and this isn't consistent with how other CLI commands do it. I guess I'd rather either 1) all CLI commands do this kind of validation or 2) all CLI commands don't, and 2) seems simplest here. What do you think?

Reply:
@epicfaace good point re: consistency. Option 1 is best for the user, but agreed that option 2 is simpler and maintains consistency. I will implement option 2 for now. If we want to go with option 1 later down the line, that can be a separate set of tasks. 👍

Reply:
Sounds good. Note that -- I think we can do Option 2 with an equivalent user experience to Option 1 though! So long as we modify the backend code so it throws a 404 not found with a specific error about bundle store not found

Reply:
Yes, we can just wrap `local.model.get_bundle_store` (in [this line](https://github.com/codalab/codalab-worksheets/blob/master/codalab/rest/bundles.py#L1109)) with a try-except in which if there is a not found error, we throw a 404.